frontend/src/metabase/dashboard/actions/save.js

```diff
@@ -150,15 +116,14 @@ export const saveDashboardAndCards = createThunkAction(
           series: dc.series,
           visualization_settings: dc.visualization_settings,
           parameter_mappings: dc.parameter_mappings,
-        }));
-        const result = await DashboardApi.updateCards({
-          dashId: dashboard.id,
-          cards,
-        });
-        if (result.status !== "ok") {
-          throw new Error(result.status);
-        }
-      }
+        })),
+      });
+      dispatch(
```

Comment:
Would it be simpler to dispatch a UPDATE_DASHCARDS with `updatedDashCards` which will update entire dashcards in the state instead of trying to update ids only??

Reply:
Yeah I think that would be simpler, it will also help later when removing tabs as it will make it easy to update the dashcards in the removed tab (either delete them or update their tab id). I will make this change.

Reply:
After looking into it I don't think we can do this. Here we are updating the `dashcardData` slice, not `dashcards`, and which are two different types. Since we don't get the `dashcardData` objects in the api response, the best we can do is map between the old ids and the new ids like are doing here.

<img width="552" alt="Screenshot 2023-04-12 at 1 07 33 PM" src="https://user-images.githubusercontent.com/37751258/231573574-42b3ec0d-e1d8-4c86-a910-a5c5edeea59b.png">

Reply:
@EmmadUsmani makes sense, thanks for looking at it anyway